pkg/compliance/checks/process_check.go

```diff
@@ -70,6 +75,21 @@ func checkProcess(e env.Env, id string, res compliance.Resource, expr *eval.Iter
 		return nil, err
 	}
 
+	if res.Fallback != nil {
```

Comment:
So each check still needs to evaluate the fallback condition? It'd be nice if we can put that in `resource_check.go`.
Another remark: should we evaluate fallback condition if check passed?

Reply:
On your first question - I was hoping to introduce it in checks where it's needed but you are right maybe it's worth moving it to `resourceCheck` as the central point where we have fallback handling. This might need changes to returns of `checkFile`, `checkAudit` etc. because we need an evaluated instance to check for the fallback condition. Let me see if that's complicated or not.

Regarding the second point - yes, even if there's a fallback and we need to use it we need to ensure we are able to resolve the primary resource and it's not an error. The result of the evaluated condition is discarded however.

Reply:
As an update - this is changed now - we check fallback condition before running the primary condition for a resource.